### PR TITLE
[WIP] First iteration of command line login.

### PR DESCRIPTION
**Pull request by wprl**
Work in progress.  Early feedback appreciated.

Adds `clinic login` command which prompts user for credentials and then fetches tokens from Cognito using `ANDMIN_NO_SRP_AUTH` flow.

Still to do:

- [ ] finish tests
- [ ] use a "real" Cognito user pool instead of test one
- [ ] fix merge issues

### CODE REVIEW

File: lib/token-util.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+// const userInfoEndpoint = `${baseUrl}/oauth2/userInfo`
+
+module.exports = { clientId, loadTokens, refreshTokens, removeTokens, saveTokens }
+
+function loadTokens (cb) {
+  fs.readFile(dotFileName, function (err, text) {
+    if (err) return cb(err)
+
+    try {
+      cb(null, JSON.parse(text))
```
Comment (mafintosh) [inline lib/token-util.js]: don't try-catch the cb. might end up calling the cb twice if the cb itself throws

File: lib/token-util.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    } catch (err) {
+      const { code } = err
+      // If the file was not there, return an empty object.
+      if (code === 'ENOENT') return cb(null, {})
```
Comment (mafintosh) [inline lib/token-util.js]: no need for destructuring `if (err.code === )`

File: lib/token-util.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+      cb(err)
+    }
+  })
+}
+
+function refreshTokens (cb) {
+  loadTokens(function (err, { refreshToken }) {
```
Comment (mafintosh) [inline lib/token-util.js]: does the destructuring here work when there is an error?

Comment (wprl) [reply lib/token-util.js]: Thanks for the feedback.  Will reopen this in clinic-private

**Pull request closed without merging**